crates/bevy_ecs/src/system/function_system.rs

```diff
@@ -108,11 +111,99 @@ impl SystemMeta {
 
     /// Marks the system as having deferred buffers like [`Commands`](`super::Commands`)
     /// This lets the scheduler insert [`apply_deferred`](`crate::prelude::apply_deferred`) systems automatically.
+    #[inline]
     pub fn set_has_deferred(&mut self) {
         self.has_deferred = true;
     }
 }
 
+impl SystemMeta {
+    /// Changes the warn policy.
+    #[inline]
+    pub fn set_param_warn_policy(&mut self, warn_policy: ParamWarnPolicy) {
```

Comment:
Do these need to be `pub`?  Most of the `SystemMeta` stuff is `pub (crate)`.  It looks like `try_warn_param` needs to be `pub` to support warning from third-party `SystemParam` implementations, but I would expect `set_param_warn_policy` and `advance_param_warn_policy` to only be called from within the crate.